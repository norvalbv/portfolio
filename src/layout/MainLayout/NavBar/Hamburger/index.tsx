import React, { ReactElement } from 'react';

type HamburgerProps = {
  onclick?: () => void;
};

const Hamburger = ({ onclick }: HamburgerProps): ReactElement => {
  const handleClick = (): void => {
    if (!onclick) return;
    onclick();
  };
  return (
    <button
      type="button"
      className="ml-3 inline-flex h-10 w-10 items-center justify-center rounded-lg p-2 text-sm focus:outline-none"
      aria-controls="navbar-hamburger"
      aria-expanded="false"
      onClick={handleClick}
    >
      <span className="sr-only">Open main menu</span>
      <svg
        className="h-5 w-5"
        aria-hidden="true"
        xmlns="http://www.w3.org/2000/svg"
        fill="none"
        viewBox="0 0 17 14"
      >
        <path
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          d="M1 1h15M1 7h15M1 13h15"
        />
      </svg>
    </button>
  );
};

export default Hamburger;
