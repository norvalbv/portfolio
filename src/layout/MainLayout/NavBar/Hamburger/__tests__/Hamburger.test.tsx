import React from 'react';
import renderer from 'react-test-renderer';

import Hamburger from '..';

describe('<Hamburger />', () => {
  test('should render correctly with mandatory props', () => {
    const tree = renderer.create(<Hamburger onclick={jest.fn()} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
