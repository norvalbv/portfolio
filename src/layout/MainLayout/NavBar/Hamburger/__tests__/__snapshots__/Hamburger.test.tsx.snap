// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hamburger /> should render correctly with mandatory props 1`] = `
<button
  aria-controls="navbar-hamburger"
  aria-expanded="false"
  className="ml-3 inline-flex h-10 w-10 items-center justify-center rounded-lg p-2 text-sm focus:outline-none"
  onClick={[Function]}
  type="button"
>
  <span
    className="sr-only"
  >
    Open main menu
  </span>
  <svg
    aria-hidden="true"
    className="h-5 w-5"
    fill="none"
    viewBox="0 0 17 14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 1h15M1 7h15M1 13h15"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
    />
  </svg>
</button>
`;
